src/main/scala/zio/package.scala

```diff
@@ -1,16 +1,30 @@
 package zio
 
+import zio.clock.Clock
+import zio.console.Console
+import zmx.server.{ ZMXConfig, ZMXServer }
+
 package object zmx extends MetricsDataModel with MetricsConfigDataModel {
   type Diagnostics = Has[Diagnostics.Service]
 
   object Diagnostics {
-    trait Service {}
+    trait Service {
+      def start: Managed[Exception, ZMXServer] //TODO refine exception type
```

Comment:
No need for `start` since it can be started inside the layer itself.

Reply:
I'm not sure I follow. So do we need the `Service` here? How would the dependency on Diagnostics be expressed and how it would be fulfilled with the layer?

Reply:
There doesn't need to be any "expressed requirement". Rather, if you add the layer to your application, then it starts up the service; otherwise, not. No need for any functions or for any accessors.

Reply:
Ok, I think I get it now. So does it make sense to keep the `Service` trait? I can make it work with this:
```
type Diagnostics = Has[Fiber.Runtime[Throwable, ZMXServer]]

def live(host: String, port: Int): ZLayer[Clock with Console, Throwable, Diagnostics] =
      ZLayer.fromManaged(ZManaged.make(ZMXServer(ZMXConfig(host, port, true)))(_.shutdown.orDie).fork)
```
Not sure the `fork` is needed, but I don't have any other idea how to run the layer in parallel to the main fiber. Also can't managed resources be run as deamons? I don't see `forkDaemon` on ZManaged.

Reply:
I think we should keep the service trait, it has a nice clean type tag and matches the pattern with all other ZIO layers.

You won't need fork daemon here because when the managed is closed, the forked fiber will be closed; and the managed will be closed at the "end of the world", e.g. `myApp.provideLayer(xyz)`.